### PR TITLE
Add windows installation, IIS reverse proxy installation and update contribute instructions

### PR DESCRIPTION
**Pull request by Faeb35**
I added a manual for installation under windows.

Additionaly I also added a manual how to setup IIS as reverse proxy by using the URL Rewrite module.

During setup git I found some inconsistencies in the [contribute readme](https://www.reddit.com/r/airsonic/comments/g8g70j/confusion_about_airsonicgithubio_repo/), which I updated as well.

Please doublecheck the texts before merging to master, as English is not my first language...

### CODE REVIEW

**Comment by Faeb35** [comment]
Remark: This pull request was requested by users here: https://github.com/airsonic/airsonic/issues/1570 
I hope it will bring some more windows users to Airsonic. Also I hope to address some common asked issues about network permissions...

File: install/windows.md [hunk 1]
```diff
@@ -0,0 +1,76 @@
+---
+layout: docs
+title: Windows Installation
+permalink: /docs/install/windows/
+---
+This page describes how to install Airsonic under Windows as Stand-alone WAR, without using Tomcat. It also contains information about how to setup Airsonic as a Windows Service and let it run under an own user/service account.
```
Comment (jvoisin) [inline install/windows.md:6]: no need for `own`

File: install/windows.md [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+
+If you like to run Airsonic as a Windows Service, download the [Non-Sucking Service Manager (NSSM)](https://nssm.cc/download).
+
+##### Install Airsonic
+
+Before we start, doublecheck if Java is working as expected and you setup the `PATH` environment variable right. Open a command prompt or the PowerShell console and type `java -version`. This should print the installed Java version.
+
+Create a new directory `C:\Program Files\Airsonic` and move the `airsonic.war` to this folder.
+
+Create a `airsonic.cmd` file in this folder and add the following content.
```
Comment (jvoisin) [inline install/windows.md:24]: *an

**Review by jvoisin — APPROVED** [review]

File: install/windows.md [hunk 3]
```diff
@@ -0,0 +1,76 @@
+---
+layout: docs
+title: Windows Installation
+permalink: /docs/install/windows/
+---
+This page describes how to install Airsonic under Windows as Stand-alone WAR, without using Tomcat. It also contains information about how to setup Airsonic as a Windows Service and let it run under an user/service account.
+
+> **NOTE**: If you like to run Airsonic by using HTTPS, take a look at the proxy section about [IIS](/docs/proxy/iis/).
```
Comment (fxthomas) [inline install/windows.md:8]: `you [would] like to run Airsonic [with] HTTPS` sounds better english :wink:

File: install/windows.md [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+
+Download the latest [Airsonic Release](https://github.com/airsonic/airsonic/releases), you need the `airsonic.war`.
+
+Install [OpenJDK](https://openjdk.java.net) or another Java implementation. If you're using vanilla Airsonic, make sure you use version 13 or below. For Airsonic Advanced with 11.0.0 or newer you need to have at least version 11, but 14 is fine as well. Make sure to add Java to your `PATH` variable.
+
+If you like to run Airsonic as a Windows Service, download the [Non-Sucking Service Manager (NSSM)](https://nssm.cc/download).
+
+##### Install Airsonic
+
+Before we start, doublecheck if Java is working as expected and you setup the `PATH` environment variable right. Open a command prompt or the PowerShell console and type `java -version`. This should print the installed Java version.
```
Comment (fxthomas) [inline install/windows.md:20]: `is working as expected and [...] the PATH environment variable [is setup correctly]` sounds better in English?

File: install/windows.md [hunk 5]
```diff
@@ -0,0 +1,76 @@
...
+
+Create a new directory `C:\Program Files\Airsonic` and move the `airsonic.war` to this folder.
```
Comment (fxthomas) [inline install/windows.md:22]: `move the airsonic.war [file] to this folder`

File: install/windows.md [hunk 6]
```diff
@@ -0,0 +1,76 @@
...
+
+Set the following settings in the `Application` tab.
+
+- Path = `java.exe`
+- Startup directory = `C:\Program Files\Airsonic`
+- Arguments = `-Dairsonic.home="C:\\Program Files\\Airsonic" -Dserver.port=4040`
+
+Open the `Shutdown` tab and remove all the flags in the checkboxes excepted the one for `Generate Contral-C`. Set the timeout to `300000`.
+
+> **NOTE**: This very long timeout (5 minutes) is needed, as Airsonic takes some time to shutdown properly.
```
Comment (fxthomas) [inline install/windows.md:52]: (off-topic) wondering why Airsonic might take 5 minutes to shut down? I don't run Windows, but on my machine Airsonic never takes more than a few seconds at most to shut down?

File: install/windows.md [hunk 7]
```diff
@@ -0,0 +1,76 @@
...
+
+That's it, you're ready to start the Windows Service by typing `net start Airsonic` in your command prompt or PowerShell console.
+
+Wait some seconds and browse to `http://localhost:4040`. You should see the Airsonic logon page.
+
+##### Run the Airsonic Windows Service under a Service Account
+
+The Airsonic Windows Service we just setup, is now running under the `SYSTEM` account. This is just fine, if all your media are on a local disk. But if you have your media on a network share, e.g. on a NAS you need to be able to connect to this share. To access this share you need to have the propper permissions. To achieve that, you should run your Airsonic under a service account and not by using the `SYSTEM` account. In this instructions this service account is called `sa-airsonic`.
```
Comment (fxthomas) [inline install/windows.md:60]: :+1: from my understanding `SYSTEM` is very close to being an admin on Windows right? If so, running as another user is a very good thing!

File: install/windows.md [hunk 8]
```diff
@@ -0,0 +1,76 @@
...
+
+> **NOTE**: Of course you can choose another name then `sa-airsonic` if you like.
+
+If you have an Active Directory domain you can create a new user in the Active Directoy with the name `sa-airsonic`. Afterwards you have to grant this user permissions on you network share/drive.
+
+If you don't have an Active Directory and running your computers in a workgroup (non-domain environment), you need to create a new Windows user called `sa-airsonic`. Make sure to create this user account with the same name and password on your device with the network share as well.
+
+The `sa-airsonic` user don't necessarily has to get the `Administrators` group/permissions, the `Users` group is just fine.
```
Comment (fxthomas) [inline install/windows.md:68]: don't has -> doesn't have

File: proxy/iis.md [hunk 9]
```diff
@@ -0,0 +1,69 @@
+---
+layout: docs
+title: Setting up IIS
+permalink: /docs/proxy/iis/
+---
+
+> **NOTE**: This manual was made by using Windows Server 2012 with IIS 8. For other versions it might be slightly different.
+
+> **NOTE**: This manual was made by running Airsonic NOT under a context.
```
Comment (fxthomas) [inline proxy/iis.md:9]: What does that mean? Maybe add a link to the MS docs about what a context is?

**Review by fxthomas — APPROVED** [review]
Nice, thank you for the contribution, that was definitely missing from the docs! I corrected a few minor things in English as requested, but nothing major, it's in fact very well-written.

File: install/windows.md (hunk 6 above)
Comment (Faeb35) [reply install/windows.md:52]: Seems to be a bug I think, it really takes a very long time. The log always says "2020-05-03 16:05:50.223  INFO --- o.a.player.service.UPnPService           : Disabling UPnP/DLNA media server" and the airsonic.lck (HSQLDB) file is removed after some seconds, but the java.exe is still running for some more minutes...
I'll try to do more investigation about this and create an issue (or a PR) if I know more about it.
For now I'll let the manual is it is, if that's fine.

File: install/windows.md (hunk 7 above)
Comment (Faeb35) [reply install/windows.md:60]: Yes that's true! I also added a line about this:
_Running the Airsonic Windows Service using another (non-admin) user also improves the safety of your system, as the `SYSTEM` account has way more rights then really needed to run Airsonic._

File: proxy/iis.md (hunk 9 above)
Comment (Faeb35) [reply proxy/iis.md:9]: Ohh, with context I meant the Airsonic context not MS.

No context: http://airsonic.public.url 
Context: http://airsonic.public.url/**context**

Is there a way to describe this clearer?

File: install/windows.md (hunk 6 above)
Comment (Faeb35) [reply install/windows.md:52]: Seems to be related to DLNA. After disabling DLNA, Airsonic respectifly the java.exe closes within seconds. I'm going to update the manual and let this part about the timeout away.

Comment (fxthomas) [reply install/windows.md]: Uh, Cling (the library Airsonic uses for DLNA) acting up again... :disappointed:

File: proxy/iis.md (hunk 9 above)
Comment (fxthomas) [reply proxy/iis.md:9]: Ah, yes, that wasn't obvious, I thought it was some MS/IIS-specific thing. I generally refer to it as the "Airsonic context path" (potentially with a link to https://airsonic.github.io/docs/configure/standalone/ where this is documented).

**Comment by fxthomas** [comment]
Nice, thank you for the help and the few changes! Merging now!

File: proxy/iis.md (hunk 9 above)
Comment (fxthomas) [reply proxy/iis.md:9]: Updated in f87b609, and updated the website in airsonic/airsonic.github.io@15b05ea011db63d4f98f9931424cb3365c25af89. Should be live soon!

**Comment by Faeb35** [comment]
Awesome!!
Thank you for the update and merging!

**Comment by antoine-airsonic** [comment]
Hello guys,

I dont know where to post so i am sharing here.
I have encountered issues for IIS to work properly on Windows 10.
I think it would be interesting to add this in the "Installation" part of the "airsonic-docs / proxy / IIS.md" documentation :


You have to download the package "Application Request Routing" on the site of IIS :
https://www.iis.net/downloads/microsoft/application-request-routing
Routing application requests (ARR) 3 is essential for the proper functioning of the script below.

**Pull request merged**